velox/functions/sparksql/tests/MapFromEntriesTest.cpp

```diff
@@ -0,0 +1,459 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <cstdint>
+#include <optional>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/lib/CheckDuplicateKeys.h"
+#include "velox/functions/prestosql/ArrayConstructor.h"
+#include "velox/functions/sparksql/tests/SparkFunctionBaseTest.h"
+#include "velox/vector/tests/TestingDictionaryArrayElementsFunction.h"
+
+using namespace facebook::velox::test;
+
+namespace facebook::velox::functions::sparksql::test {
+namespace {
+std::optional<std::vector<std::pair<int32_t, std::optional<int32_t>>>> O(
+    const std::vector<std::pair<int32_t, std::optional<int32_t>>>& vector) {
```

Comment:
The function name lacks information. Could you use a better one?

Reply:
+1